src/code42cli/cmds/devices.py

```diff
@@ -0,0 +1,383 @@
+from collections import OrderedDict
+from datetime import date
+from datetime import datetime
+
+import click
+from pandas import DataFrame
+from pandas import to_datetime
+from pandas import to_timedelta
+from py42 import exceptions
+from py42.exceptions import Py42NotFoundError
+
+from code42cli.bulk import run_bulk_process
+from code42cli.click_ext.groups import OrderedGroup
+from code42cli.errors import Code42CLIError
+from code42cli.file_readers import read_csv_arg
+from code42cli.options import format_option
+from code42cli.options import sdk_options
+from code42cli.output_formats import DataFrameOutputFormatter
+from code42cli.output_formats import OutputFormatter
+
+
+@click.group(cls=OrderedGroup)
+@sdk_options(hidden=True)
+def devices(state):
+    """For managing devices within your Code42 environment."""
+    pass
+
+
+device_guid_argument = click.argument("device-guid", type=str)
+
+change_device_name_option = click.option(
+    "--change-device-name",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="""Prepend "deactivated_" and today's date to the name of any
+    deactivated devices.""",
+)
+
+purge_date_option = click.option(
+    "--purge-date",
+    required=False,
+    type=str,
+    default=None,
+    help="""The date on which the archive should be purged from cold storage in yyyy-MM-dd format.
+    If not provided, the date will be set according to the appropriate org settings.""",
+)
+
+
+@devices.command()
+@device_guid_argument
+@change_device_name_option
+@purge_date_option
+@sdk_options()
+def deactivate(state, device_guid, change_device_name, purge_date):
+    """Deactivate a device within Code42. Requires the device GUID to deactivate."""
+    _deactivate_device(state.sdk, device_guid, change_device_name, purge_date)
+
+
+def _deactivate_device(sdk, device_guid, change_device_name, purge_date):
+    device = sdk.devices.get_by_guid(device_guid)
+    try:
+        sdk.devices.deactivate(device.data["computerId"])
+    except exceptions.Py42BadRequestError:
+        raise Code42CLIError("The device {} is in legal hold.".format(device_guid))
+    except exceptions.Py42NotFoundError:
+        raise Code42CLIError("The device {} was not found.".format(device_guid))
+    except exceptions.Py42ForbiddenError:
+        raise Code42CLIError("Unable to deactivate {}.".format(device_guid))
+    if purge_date or change_device_name:
+        guid = device_guid
+        name = device.data["name"]
```

Comment:
This seems unnecessary, at least the `if purge_date: guid = device_guid` portion.